python/social_bot/teacher.py

```diff
@@ -232,6 +227,10 @@ def __init__(self, task_groups_exclusive=True):
         """
         self._task_groups_exclusive = task_groups_exclusive
         self._vocab_list = None
+        self._task_groups = []
```

Comment:
In the testing "_task_groups" won't free, and keeps increasing during the whole test, so move them to __init__()